fhir-path/src/commonMain/kotlin/com/google/fhir/fhirpath/FhirPathEvaluator.kt

```diff
@@ -61,6 +61,7 @@ internal class FhirPathEvaluator(initialContext: Any?) : fhirpathBaseVisitor<Col
 
   private val contextStack = ArrayDeque<Collection<Any>>()
   private val thisStack = ArrayDeque<Any>()
+  private val totalStack = ArrayDeque<Collection<Any>>()
```

Comment:
is the totalStack ever more than 1 item? If not, we can just make this a single item as cache?

but - maybe you need to consider recursion (an aggregate function inside another aggregate function), in that case maybe this is legitimate.